Code/Editor/ViewPane.cpp

```diff
@@ -142,65 +157,287 @@ class ViewportTitleExpanderWatcher
+    }
+    {
+        AzToolsFramework::MenuProperties menuProperties;
+        menuProperties.m_name = "Viewport Debug Info";
+        m_menuManagerInterface->RegisterMenu(ViewportDebugInfoMenuIdentifier, menuProperties);
+    }
+    {
+        AzToolsFramework::MenuProperties menuProperties;
+        menuProperties.m_name = "Viewport Helpers";
+        m_menuManagerInterface->RegisterMenu(ViewportHelpersMenuIdentifier, menuProperties);
+    }
+}
+
+void CLayoutViewPane::OnToolBarRegistrationHook()
+{
+    // Register top viewport toolbar.
+    AzToolsFramework::ToolBarProperties toolBarProperties;
+    toolBarProperties.m_name = "Viewport ToolBar";
+    m_toolBarManagerInterface->RegisterToolBar("o3de.toolbar.viewport.top", toolBarProperties);
+
+    // Add toolbar to top of viewport.
+    QToolBar* toolBar = m_toolBarManagerInterface->GetToolBar("o3de.toolbar.viewport.top");
+    addToolBar(Qt::TopToolBarArea, toolBar);
+}
+
+void CLayoutViewPane::OnActionRegistrationHook()
+{
+    // Viewport Debug Information
+    {
+        constexpr AZStd::string_view actionIdentifier = "o3de.action.viewport.info.toggle";
+        AzToolsFramework::ActionProperties actionProperties;
+        actionProperties.m_name = "Toggle Between States";
+        actionProperties.m_category = "Viewport Debug Information";
+        actionProperties.m_iconPath = ":/Menu/debug.svg";
+
+        m_actionManagerInterface->RegisterCheckableAction(
+            EditorMainWindowActionContextIdentifier,
+            actionIdentifier,
+            actionProperties,
+            [viewportTitleDlg = m_viewportTitleDlg]
+            {
+                viewportTitleDlg->OnToggleDisplayInfo();
+            },
+            []() -> bool
+            {
+                AZ::AtomBridge::ViewportInfoDisplayState currentState = AZ::AtomBridge::ViewportInfoDisplayState::NoInfo;
+                AZ::AtomBridge::AtomViewportInfoDisplayRequestBus::BroadcastResult(
+                    currentState, &AZ::AtomBridge::AtomViewportInfoDisplayRequestBus::Events::GetDisplayState);
+
+                return currentState != AZ::AtomBridge::ViewportInfoDisplayState::NoInfo;
+            }
+        );
+
+        m_actionManagerInterface->AddActionToUpdater(ViewportDisplayInfoStateChangedUpdaterIdentifier, actionIdentifier);
+    }
+    {
+        constexpr AZStd::string_view actionIdentifier = "o3de.action.viewport.info.normal";
+        AzToolsFramework::ActionProperties actionProperties;
+        actionProperties.m_name = "Normal";
+        actionProperties.m_category = "Viewport Debug Information";
+
+        m_actionManagerInterface->RegisterCheckableAction(
+            EditorMainWindowActionContextIdentifier,
+            actionIdentifier,
+            actionProperties,
+            [viewportTitleDlg = m_viewportTitleDlg]
+            {
+                viewportTitleDlg->SetNormalViewportInfo();
+            },
+            []() -> bool
```

Comment:
nit: the `-> bool` should be redundant and can be omitted if desired

Reply:
Yeah I tend to leave it by force of habit, and I like that it makes it more readable on which one is the behavior and which one is the callback.